src/connectedk8s/azext_connectedk8s/custom.py

```diff
@@ -1026,11 +1029,19 @@ def update_connected_cluster(cmd, client, resource_group_name, cluster_name, htt
     if os.getenv('HELMREPONAME') and os.getenv('HELMREPOURL'):
         utils.add_helm_repo(kube_config, kube_context, helm_client_location)
 
-    # Setting the config dataplane endpoint
-    config_dp_endpoint = get_config_dp_endpoint(cmd, connected_cluster.location)
+    # Read and validate the helm values file for Dogfood.
+    config_dp_endpoint = None
+    if cmd.cli_ctx.cloud.endpoints.resource_manager == consts.Dogfood_RMEndpoint:
+        config_dp_endpoint, release_train_custom = validate_env_file_dogfood(values_file, values_file_provided)
+    # Get the values or endpoints required for retreiving the Helm registry URL.
+    if "dataplaneEndpoints" in metadata:
```

Comment:
can the check '    if "dataplaneEndpoints" in metadata:'  be part of the method get_config_dp_endpoint, so that it need not be repated?